appcues/src/main/java/com/appcues/statemachine/Error.kt

```diff
@@ -3,17 +3,18 @@ package com.appcues.statemachine
 import com.appcues.data.model.Experience
 import java.util.UUID
 
-internal sealed class Error(open val message: String, errorId: UUID? = null) {
+internal sealed class Error(open val message: String) {
+    private val _id = UUID.randomUUID()
 
-    val id: UUID = errorId ?: UUID.randomUUID()
+    // allows for override of error ID with a value driven by the experience.renderErrorId
+    // for matching error/recovery IDs on render retry
+    var errorId: UUID? = null
```

Comment:
i guess we could also do val id:: UUID = UUID.randomUUID() ?

Reply:
all errors could have a UUID wouldnt hurt. for simplicity

Reply:
yeah all errors do have an ID, it is just that sometimes we need to override the ID with a "render error ID" that matches that value from the experience that gets set. That way, we can create another recovery event with a matching error ID - to tie the error to the recovery correctly.